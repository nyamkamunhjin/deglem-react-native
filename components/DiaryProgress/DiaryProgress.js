import React, { useEffect, useState } from 'react';
import { View, StyleSheet } from 'react-native';
import * as Progress from 'react-native-progress';
import { Text, withTheme } from 'react-native-paper';

/**
 * @author
 * @function DiaryProgress
 **/
const DiaryProgress = ({ theme, progress}) => {
  const { colors, fonts } = theme;
  const { container, circles } = styles;



  return (
    <View style={container}>
      <Progress.Circle
        // indeterminate={true}
        style={circles}
        progress={progress || 0}
        size={150}
        strokeCap={'round'}
        allowFontScaling={true}
        formatText={() => (
          <Text style={{ fontFamily: fonts.light.fontFamily, fontSize: 15 }}>
            567/2500
          </Text>
        )}
        showsText={true}
        thickness={5}
        animated={true}
        color={colors.triadic}
        // unfilledColor={colors.disabled}
        borderWidth={0}
        endAngle={0.5}
        // fill={colors.triadic}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    margin: 5,
    borderRadius: 10,
  },
  circles: {
    margin: 10,
    // height: 500
  },
});
export default withTheme(DiaryProgress);
