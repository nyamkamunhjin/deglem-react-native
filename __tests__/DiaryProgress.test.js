/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import { render, fireEvent } from '@testing-library/react-native';
import DiaryProgress from '../components/DiaryProgress/DiaryProgress';

describe('components', () => {
  test('DiaryProgress render test', async () => {
    const tree = render(<DiaryProgress />);
    // console.log(tree.getInstance());
    expect(tree.toJSON()).toMatchSnapshot();
    // expect(tree.getInstance().test(2)).toBe(4);
  });
});
