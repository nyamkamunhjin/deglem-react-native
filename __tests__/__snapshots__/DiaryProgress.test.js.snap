// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components DiaryProgress render test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 10,
      "justifyContent": "center",
      "margin": 5,
      "textAlign": "center",
    }
  }
>
  <View
    indeterminateAnimationDuration={1000}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
        },
        Object {
          "margin": 10,
        },
      ]
    }
  >
    <ARTSurfaceView
      style={
        Array [
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          },
          Object {
            "height": 150,
            "width": 150,
          },
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            75,
            2.5,
            2,
            75,
            2.5,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#007aff"
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={5}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTSurfaceView>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 70,
          "height": 140,
          "justifyContent": "center",
          "left": 5,
          "position": "absolute",
          "top": 5,
          "width": 140,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "rgba(0, 122, 255, 1)",
              "fontSize": 31.11111111111111,
              "fontWeight": "300",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Object {
                "fontFamily": "System",
                "fontSize": 15,
              },
            ]
          }
        >
          567/2500
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
